### PR TITLE
godot-cpp: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by RubenRBS — CHANGES_REQUESTED** [review]
blocking, found something strange

File: recipes/godot-cpp/all/conanfile.py [hunk 1]
```diff
@@ -12,20 +20,81 @@ class GodotCppConan(ConanFile):
...
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires(f"godot_headers/{self.version}", transitive_headers=True)
+
+    def package_id(self):
+        if self.info.settings.build_type != "Debug":
+            self.info.settings.build_type = "Release"
```
Comment (RubenRBS) [inline recipes/godot-cpp/all/conanfile.py:45]: Why is this needed? https://github.com/godotengine/godot-cpp/blob/master/CMakeLists.txt#L2 seems to imply this is an option. Does it actually always produce the same resulting binaries?

**Review by RubenRBS — COMMENTED** [review]
Question about the package_id modification :)

File: recipes/godot-cpp/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/godot-cpp/all/conanfile.py:45]: `RelWithDebInfo` and other less common build types get built as `Release` by the project, so they are all equivalent. `Debug` is the only exception.

Comment (RubenRBS) [reply recipes/godot-cpp/all/conanfile.py:45]: Oh I missed that its looking for soemthing _other than_ debug  missed the negation. Thanks for clarifying

**Review by RubenRBS — APPROVED** [review]
